### PR TITLE
Show previous user version purl links

### LINKED ISSUE #3582: Add links to item view page for previous user versions of a work
This includes the following changes to the Details table at the top of an item view page:

- change existing "Version details" label to say "Current version". Note that the content displayed in the second column is not changing. 
- add a new row to the table below the "Current version" row with the label "Previous version(s)"
- display a list of numbers for the previous versions, starting with the most recent and counting back to 1, separated by commas. Each number should be linked to the PURL for that specific version

Screenshot 2024-07-24 at 2 14 41 PM

### PR DESCRIPTION
**Pull request by lwrubel**
# Why was this change made? 🤔
Resolves #3582 to show links to purls for previous versions. Does not change the current prod view where user versions are not enabled. HOLD for PO review.

When there is only one user version, no Previous versions row displays:

Screenshot 2024-07-26 at 10 31 14 AM

When previous user versions exist: 

Screenshot 2024-07-26 at 10 35 03 AM

# How was this change tested? 🤨
Unit and testing on QA.

### CODE REVIEW

File: app/components/works/detail_component.rb
Comment (justinlittman) [inline app/components/works/detail_component.rb:110]: L110-112 could just be `work_version.user_version > 1`

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of comment and approval by Amy.

File: app/components/works/detail_component.rb
Comment (lwrubel) [reply app/components/works/detail_component.rb]: Fixed, thanks.

**Comment by lwrubel** [comment]
Merging after @amyehodge tested H2 and approved.

**Pull request merged**
